tx-parser/src/lib.rs

```diff
@@ -30,181 +34,150 @@ fn fn_parser<O, I>(f: fn(I) -> ParseResult<O, I>, err: &'static str) -> TxParser
     parser(f).expected(err)
 }
 
-impl<I> Tx<I>
-    where I: Stream<Item = Value>
-{
-    fn integer() -> TxParser<i64, I> {
-        fn_parser(Tx::<I>::integer_, "integer")
-    }
+def_parser_fn!(Tx, integer, Value, i64, input, {
+    return satisfy_map(|x: Value| if let Value::Integer(y) = x {
+            Some(y)
+        } else {
+            None
+        })
+        .parse_stream(input);
+});
 
-    fn integer_(input: I) -> ParseResult<i64, I> {
-        return satisfy_map(|x: Value| if let Value::Integer(y) = x {
-                Some(y)
-            } else {
-                None
-            })
-            .parse_stream(input);
-    }
+// The above using a def_value_satisfy_parser_fn
+//def_value_satisfy_parser_fn!(Tx, integer, i64, if let Value::Integer(y) = x { Some(y) } else { None });
```

Comment:
`i64` implements `Copy`, so we should fix the definition of `as_integer` (and `as_float`, `as_boolean`). No sense returning a reference type when the raw type is no larger than a machine address!

I can do that very quickly in a side PR. Momento…